### PR TITLE
[MI-152] Help Center Categories

### PR DESCRIPTION
**Pull request by joseconsador**
Change directory and namespace structure.
Add Help Center Categories resource and tests.

/cc @miogalang @jwswj @mmolina @atroche

### References
 - Jira link: https://zendesk.atlassian.net/browse/MI-152

### Risks
 - Medium - the client might not be able to access a categories endpoint with the correct locale

### CODE REVIEW

File: src/Zendesk/API/HttpClient.php [hunk 1]
```diff
@@ -122,13 +123,17 @@ class HttpClient
     protected $sideload;
 
     /**
-     * @var Debug
+     * @var debug
```
Comment (miogalang) [inline src/Zendesk/API/HttpClient.php]: Why?

File: src/Zendesk/API/Resources/HelpCenter.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+use Zendesk\API\Traits\Utility\ChainedParametersTrait;
+use Zendesk\API\Traits\Utility\InstantiatorTrait;
+
+/**
+ * This class serves as a container to allow $this->client->helpCenter
+ *
+ * @method Categories categories()
+ */
+class HelpCenter
+{
```
Comment (miogalang) [inline src/Zendesk/API/Resources/HelpCenter.php]: Please add docbocks to the methods

File: src/Zendesk/API/Resources/HelpCenter/Categories.php [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+use Zendesk\API\Traits\Resource\Defaults;
+
+class Categories extends ResourceAbstract
+{
+    const OBJ_NAME = 'category';
+    const OBJ_NAME_PLURAL = 'categories';
+
+    use Defaults;
+
+    private $locale;
```
Comment (miogalang) [inline src/Zendesk/API/Resources/HelpCenter/Categories.php]: Please add docblock here too

File: src/Zendesk/API/Resources/HelpCenter/ResourceAbstract.php [hunk 4]
```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace Zendesk\API\Resources\HelpCenter;
+
+class ResourceAbstract extends \Zendesk\API\Resources\ResourceAbstract
```
Comment (miogalang) [inline src/Zendesk/API/Resources/HelpCenter/ResourceAbstract.php]: Docblocks here too

File: src/Zendesk/API/Resources/HelpCenter.php (hunk 2 above)
Comment (joseconsador) [inline src/Zendesk/API/Resources/HelpCenter.php]: :haircut:

**Comment by jwswj** [comment]
:+1:

**Comment by miogalang** [comment]
:+1:

**Pull request merged**